api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirTaskDaoImpl.java

```diff
@@ -23,45 +23,38 @@
 import lombok.AccessLevel;
 import lombok.Setter;
 import org.hibernate.Criteria;
-import org.hibernate.SessionFactory;
 import org.hibernate.criterion.Restrictions;
 import org.hl7.fhir.r4.model.DomainResource;
 import org.openmrs.api.db.DAOException;
 import org.openmrs.module.fhir2.FhirTask;
 import org.openmrs.module.fhir2.api.dao.FhirTaskDao;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Component;
 
 @Component
 @Setter(AccessLevel.PACKAGE)
-public class FhirTaskDaoImpl extends BaseDao implements FhirTaskDao {
-	
-	@Autowired
-	@Qualifier("sessionFactory")
-	SessionFactory sessionFactory;
+public class FhirTaskDaoImpl extends BaseFhirDao<FhirTask> implements FhirTaskDao {
 	
 	@Override
 	public FhirTask saveTask(FhirTask task) throws DAOException {
 		// TODO: Refactor - and figure out why CascadeType.ALL does not take care of this.
 		if (task.getOwnerReference() != null && task.getOwnerReference().getReference() != null) {
-			sessionFactory.getCurrentSession().saveOrUpdate(task.getOwnerReference());
+			getSessionFactory().getCurrentSession().saveOrUpdate(task.getOwnerReference());
 		}
 		
-		sessionFactory.getCurrentSession().saveOrUpdate(task);
+		getSessionFactory().getCurrentSession().saveOrUpdate(task);
 		
 		return task;
 	}
 	
 	@Override
-	public FhirTask getTaskByUuid(String uuid) {
-		return (FhirTask) sessionFactory.getCurrentSession().createCriteria(FhirTask.class).add(eq("uuid", uuid))
+	public FhirTask get(String uuid) {
+		return (FhirTask) getSessionFactory().getCurrentSession().createCriteria(FhirTask.class).add(eq("uuid", uuid))
 		        .uniqueResult();
```

Comment:
We shouldn't need this implementation. It should be taken care of by BaseFhirDao.